### ISSUE TITLE
Two false negatives from official document

### ISSUE THREAD

**Issue by Zustin** [google/error-prone#2476]
Hi! Recently, I found two false negatives about using when used Error Prone. After reading official document, I think errorprone should have recognized it because yhey are related to FloatingPointLiteralPrecision. 

Actually, it cannot recognize the [officical case](https://errorprone.info/bugpattern/FloatingPointLiteralPrecision).

The minimmal case is listed below:
```java
public void testFloatingPointLiteralPrecision() {
    double d = 1.9999999999999999999999999999999;  // should report a warning here
    float f = 1.99999999999999999999999f;  // should report a warning here
}
```

Expected output: two bug warnings
Actual output: No warnings and d and f are regarded as 2.0

**Comment by cushon** [comment]
I can't reproduce. Note that by default the check is a warning, not an error, so it won't fail the build.

Which build system integration are you using? Do you have a repro?

```java
package t;
class T {
  public void testFloatingPointLiteralPrecision() {
    double d = 1.9999999999999999999999999999999; // should report a warning here
    float f = 1.99999999999999999999999f; // should report a warning here
  }
}
```

```
javac \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED \
  -J--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED \
  -J--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED \
  -J--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED \
  -XDcompilePolicy=simple \
  -processorpath error_prone_core-2.8.1-with-dependencies.jar:dataflow-shaded-3.11.0.jar:jFormatString-3.0.0.jar \
  '-Xplugin:ErrorProne' \
  T.java
...
T.java:4: warning: [UnusedVariable] The local variable 'd' is never read.
    double d = 1.9999999999999999999999999999999; // should report a warning here
           ^
    (see https://errorprone.info/bugpattern/UnusedVariable)
  Did you mean to remove this line?
T.java:5: warning: [UnusedVariable] The local variable 'f' is never read.
    float f = 1.99999999999999999999999f; // should report a warning here
          ^
    (see https://errorprone.info/bugpattern/UnusedVariable)
  Did you mean to remove this line?
T.java:4: warning: [FloatingPointLiteralPrecision] Floating point literal loses precision
    double d = 1.9999999999999999999999999999999; // should report a warning here
               ^
    (see https://errorprone.info/bugpattern/FloatingPointLiteralPrecision)
  Did you mean 'double d = 2.0;'?
T.java:5: warning: [FloatingPointLiteralPrecision] Floating point literal loses precision
    float f = 1.99999999999999999999999f; // should report a warning here
              ^
    (see https://errorprone.info/bugpattern/FloatingPointLiteralPrecision)
  Did you mean 'float f = 2.0f;'?
4 warnings
```

**Comment by Zustin** [comment]
OK. Got it, thanks for your help.
